Sources/Starknet/Accounts/StarknetAccount.swift

```diff
@@ -86,100 +86,100 @@ public class StarknetAccount: StarknetAccountProtocol {
         return makeDeployAccountTransactionV3(classHash: classHash, salt: salt, calldata: calldata, signature: signature, params: params, forFeeEstimation: forFeeEstimation)
     }
 
-    public func executeV1(calls: [StarknetCall], params: StarknetOptionalInvokeParamsV1) async throws -> StarknetInvokeTransactionResponse {
+    public func executeV1(calls: [StarknetCall], params: StarknetOptionalInvokeParamsV1) async throws -> StarknetRequest<StarknetInvokeTransactionResponse, AddInvokeTransactionParams> {
         var nonce: Felt
         var maxFee: Felt
 
         if let paramsNonce = params.nonce {
             nonce = paramsNonce
         } else {
-            nonce = try await getNonce()
+            nonce = try await getNonce().send()
         }
 
         if let paramsMaxFee = params.maxFee {
             maxFee = paramsMaxFee
         } else {
-            let feeEstimate = try await estimateFeeV1(calls: calls, nonce: nonce)
+            let feeEstimate = try await estimateFeeV1(calls: calls, nonce: nonce).send()[0]
             maxFee = feeEstimate.toMaxFee()
         }
 
         let params = StarknetInvokeParamsV1(nonce: nonce, maxFee: maxFee)
         let signedTransaction = try signV1(calls: calls, params: params, forFeeEstimation: false)
 
-        return try await provider.addInvokeTransaction(signedTransaction)
+        return provider.addInvokeTransaction(signedTransaction)
     }
 
-    public func executeV3(calls: [StarknetCall], params: StarknetOptionalInvokeParamsV3) async throws -> StarknetInvokeTransactionResponse {
+    public func executeV3(calls: [StarknetCall], params: StarknetOptionalInvokeParamsV3) async throws -> StarknetRequest<StarknetInvokeTransactionResponse, AddInvokeTransactionParams> {
         var nonce: Felt
         var resourceBounds: StarknetResourceBoundsMapping
 
         if let paramsNonce = params.nonce {
             nonce = paramsNonce
         } else {
-            nonce = try await getNonce()
+            nonce = try await getNonce().send()
         }
 
         if let paramsResourceBounds = params.resourceBounds {
             resourceBounds = paramsResourceBounds
         } else {
-            let feeEstimate = try await estimateFeeV3(calls: calls, nonce: nonce)
+            let feeEstimate = try await estimateFeeV3(calls: calls, nonce: nonce).send()[0]
```

Comment:
Are you sure getting the first estimate is a valid approach?

Reply:
Has been discussed in previous thread
https://github.com/software-mansion/starknet.swift/pull/187#discussion_r1600240910